File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.java
class Solution {
    public int numberOfSubstrings(String s) {
        int result = 0;
        int[] last = {-1,-1,-1};
        int min = -1;
        for(int i = 0; i < s.length(); i++){
            int ind = s.charAt(i) - 'a';
            last[ind] = i;
            min = Integer.MAX_VALUE;
            for(int j : last)
                min = Math.min(min,j);
            result += min + 1;            


        }
        return result;


        
    }
}

File: 1358-number-of-substrings-containing-all-three-characters/1358-number-of-substrings-containing-all-three-characters.py
class Solution:
    def numberOfSubstrings(self, s: str) -> int:
        last = [-1,-1,-1]
        l = 0 
        n = len(s)
        result = 0
        for r in range(n):
            ind =  ord(s[r]) - ord("a")
            last[ind] = r
            m = min(last)
            result += m + 1
        return result

        